### DISCUSSION TITLE
2022-03-21 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1198] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2391853)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2391854)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2391855)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2391856)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2391857)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2391858)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2408203)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2408562)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2409026)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2410219)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2412416)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2413079)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/1198#discussioncomment-2415757)
## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### Feedback Requested
- https://github.com/centerofci/mathesar/discussions/1184
- https://github.com/centerofci/mathesar/discussions/1193
- https://github.com/centerofci/mathesar/discussions/1192

### Complete
- **Discussions**:
    - https://github.com/centerofci/mathesar/discussions/1154
    - https://github.com/centerofci/mathesar/discussions/1186
- **Full stack and general work**:
    - https://github.com/centerofci/mathesar/issues/1129
    - https://github.com/centerofci/mathesar/issues/1130
    - https://github.com/centerofci/mathesar/issues/1125
    - https://github.com/centerofci/mathesar/issues/1124
    - https://github.com/centerofci/mathesar/issues/1123
- **Backend work**:
    - https://github.com/centerofci/mathesar/issues/1156
    - https://github.com/centerofci/mathesar/issues/1064
    - https://github.com/centerofci/mathesar/pull/1176
- **Frontend work**:
    - https://github.com/centerofci/mathesar/issues/1161
    - https://github.com/centerofci/mathesar/issues/1054
    - https://github.com/centerofci/mathesar/issues/1118
    - https://github.com/centerofci/mathesar/issues/984
- **Documentation work**:
    - https://github.com/centerofci/mathesar/issues/845

**Comment by kgodey** [comment]
## Frontend priorities

1. Get open PRs merged:
    1. https://github.com/centerofci/mathesar/pull/1155
    1. https://github.com/centerofci/mathesar/pull/1170
    1. https://github.com/centerofci/mathesar/pull/1182
    1. https://github.com/centerofci/mathesar/pull/1191
2. Make progress on high-priority issues for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) milestone:
    1. #970
    1. #930 
    1. #1108
    1. #257 
    1. #775
3. Continue working on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) frontend issues.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good

**Reply by pavish** [reply]
Looks good to me

**Comment by kgodey** [comment]
## Backend priorities

1. Get open PRs completed and merged:
    1. https://github.com/centerofci/mathesar/pull/1174
    1. https://github.com/centerofci/mathesar/pull/1188
    1. https://github.com/centerofci/mathesar/pull/1190
    1. https://github.com/centerofci/mathesar/pull/1191
2. Complete remaining backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11):
    1. @dmos62:
        1. https://github.com/centerofci/mathesar/issues/1100
        1. https://github.com/centerofci/mathesar/issues/1037
    2. @mathemancer
        1. https://github.com/centerofci/mathesar/issues/408 
        2. https://github.com/centerofci/mathesar/issues/411 
        3. https://github.com/centerofci/mathesar/issues/414
        4. https://github.com/centerofci/mathesar/issues/384
        5. https://github.com/centerofci/mathesar/issues/429
        6. https://github.com/centerofci/mathesar/issues/419
        7. https://github.com/centerofci/mathesar/issues/432
    3. @silentninja:
        1. https://github.com/centerofci/mathesar/issues/839
        1. https://github.com/centerofci/mathesar/issues/996
        1. https://github.com/centerofci/mathesar/issues/1105
    4. Not assigned to core team members:
        1. https://github.com/centerofci/mathesar/issues/1145 (assigned to @ManishShah120)
        1. https://github.com/centerofci/mathesar/issues/1167 (unassigned)
3. Start work on backend work for [Links between Tables](https://github.com/centerofci/mathesar/milestone/47), some ideas:
    1. https://github.com/centerofci/mathesar/issues/943
    2. https://github.com/centerofci/mathesar/issues/944
    3. https://github.com/centerofci/mathesar/issues/973
    4. https://github.com/centerofci/mathesar/issues/972
4. #1008

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by mathemancer** [reply]
I think this makes sense.

**Reply by dmos62** [reply]
Makes sense.

**Comment by kgodey** [comment]
## Design priorities

1. Publish spec and start review process for #1065
2. Continue iterating on #978 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ghislaineguerin** [reply]
Looks good to me. I will continue working on #1065

**Comment by kgodey** [comment]
## Product priorities

_No product work is planned for this week._

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-03-21 to 2022-03-25 | Mukesh |
| 2022-03-28 to 2022-04-01 | Sean |
| 2022-04-04 to 2022-04-08 | Ghislaine  |
| 2022-04-11 to 2022-04-12 | Kriti |
| 2022-04-13 to 2022-04-15 | Dominykas |
| 2022-04-18 to 2022-04-22 | Brent |
| 2022-04-25 to 2022-04-29 | Kriti |
| 2022-05-02 to 2022-05-06 | Pavish |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- It's kind of crazy that my [Number component](https://github.com/centerofci/mathesar/pull/1155) is still in draft state. At this point I'm waiting for a sync call with Pavish where we can discuss the best approach to fixing a regression from my work that an E2E test uncovered. Aside from that (hopefully small) issue, the PR should be ready to bring out of draft state, which I'm very excited about. I never imagined that data entry of numbers could be so complex! But all-in-all I'm very happy with how this PR has turned out, even though it took a while.

- Since our involvement in GSoC began, I'm definitely spending significantly less time on my own code because I'm reviewing more PRs, helping more people, and following more discussions. All things considered, this seems good. But I'll have somewhat lower expectations for my own output during GSoC.

> What are you planning to work on soon?

- #930
- #1108
- #775

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Need to chat with Pavish about NumberInput, but that's on his radar and we may even be able to do that today.

**Comment by mathemancer** [comment]
> How did your most recent work go?

I had some family stuff come up that meant working while managing the baby. However, this gave me a good reason to jump into more PR reviews and documentation, which went well.

> What are you planning to work on soon?

I really want to focus more on coding this week than last, specifically on grouping.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Not at the moment.

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

The PR for updates for text and groundwork for db type determination is merged. The groundwork for display options is mostly done with some race conditions to work out. I'm working on the boolean display options, which is taking slightly more time than anticipated.

I've also been focusing on helping new contributors the last week and it's been going well.

> What are you planning to work on soon?

* Complete [boolean type implementation](https://github.com/centerofci/mathesar/issues/257)
* Setup display options for number and money (This is different from the implementation of cells)
* Start work on [data & time types implementation](https://github.com/centerofci/mathesar/issues/260), if I get the time.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

It went well, lots of discussions around Data Explorer and specs.

> What are you planning to work on soon?

#1065 Finalize Spec

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by kgodey** [comment]
>  How did your most recent work go?

I haven't gotten much focused work done in the past week, there's a lot more day-to-day activity and discussion since the GSoC contribution period started. That's fine, though. I've also been spending a fair amount of time on the query builder design with @ghislaineguerin.

> What are you planning to work on soon?

- Keep up with day to day stuff
- Continue and hopefully work on #1008 
- Keep reviewing draft work on #1065

>  Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Comment by silentninja** [comment]
>  How did your most recent work go?

Progress was really slow with the issue that I am currently working on(`Money type display options`). On the other hand, there was a good amount of discussion with GSOC contributors. 

> What are you planning to work on soon?

- Finish the remaining issues listed in the backend priorities section. 
- Help contributors with the codebase and add documentations that would be helpful to new contributors


>  Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Comment by dmos62** [comment]
>  How did your most recent work go?

Ran into some technical debt and ended up taking a deep dive into type logic related stuff in Mathesar. That was productive in that I've gotten a better understanding of our backend, and there's a pretty nice refactor under-way, but at the same time I've contributed little code during last week.

> What are you planning to work on soon?

- flesh out the mentioned refactor
- custom error debugging related issue
  - assign self or find someone else to take the task
  - https://github.com/centerofci/mathesar/issues/1151
- start working on next milestone.

>  Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Reply by kgodey** [reply]
@dmos62 Will the type refactor also resolve #1037 (by removing aliases from the API?)

**Reply by dmos62** [reply]
Yes, it will.

**Reply by kgodey** [reply]
Great, thanks!
